tiledb/sm/query/result_space_tile.h

```diff
@@ -85,19 +82,77 @@ struct ResultSpaceTile {
   /** Default move-assign operator. */
   ResultSpaceTile& operator=(ResultSpaceTile&& result_space_tile) = default;
 
+  /** Returns the fragment domains. */
+  const std::vector<std::pair<unsigned, std::reference_wrapper<const NDRange>>>&
+  frag_domains() const {
+    return frag_domains_;
+  }
+
+  /** Returns the result tiles. */
+  const std::map<unsigned, ResultTile>& result_tiles() const {
+    return result_tiles_;
+  }
+
+  /** Returns the start coordinates. */
+  const std::vector<T>& start_coords() const {
+    return start_coords_;
+  }
+
+  /** Sets the start coordinates. */
+  void set_start_coords(const std::vector<T>& start_coords) {
+    start_coords_ = start_coords;
+  }
+
+  /** Appends a fragment domain. */
+  void append_frag_domain(
+      unsigned frag_idx, const std::reference_wrapper<const NDRange>& dom) {
+    frag_domains_.emplace_back(frag_idx, dom);
+  }
+
+  /** Sets the input result tile for the given fragment. */
+  void set_result_tile(unsigned frag_idx, const ResultTile& result_tile) {
```

Comment:
Let's add an `assert(result_tile.count(frag_idx) == 0)` to ensure we're not overwriting a result tile.

Reply:
Done.